### DISCUSSION TITLE
How do i merge a pull request from github cli?

### DISCUSSION THREAD

**Discussion by TestUser123123-svg** [community/community#135788] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

How do i merge a pull request from github cli?

**Comment by Gvolexe** [comment, marked as answer]
To merge a pull request (PR) using the GitHub CLI (gh), follow these steps:

Install GitHub CLI (if not already installed):

You can download and install it from [GitHub CLI releases](https://github.com/cli/cli/releases) or via a package manager (e.g., brew install gh for macOS, or sudo apt install gh for Ubuntu).
Authenticate with GitHub:

Run gh auth login to authenticate. You can choose to authenticate via web browser or by pasting an authentication token.
Navigate to Your Repository:

If you're not already in the directory of your repository, navigate to it using:
sh
Copy code
cd path/to/your/repository
View the List of Pull Requests:

To view open pull requests, you can use:
sh
Copy code
gh pr list
Check Out the Pull Request (optional but recommended):

If you want to review the changes locally, you can check out the pull request branch:
sh
Copy code
gh pr checkout <PR-number>
Replace <PR-number> with the number of the pull request.
Merge the Pull Request:

To merge the pull request, use:
sh
Copy code
gh pr merge <PR-number>
This command will prompt you to choose a merge method (merge, squash, rebase). You can specify the method directly with a flag:
Merge: --merge
Squash and Merge: --squash
Rebase and Merge: --rebase
Example:
sh
Copy code
gh pr merge <PR-number> --merge
If you want to skip the confirmation prompt, you can add --auto:
sh
Copy code
gh pr merge <PR-number> --merge --auto
Push the Changes (if required):

If you opted to checkout and test the PR locally, remember to push the merged changes back to the main branch, if not automatically done by the gh pr merge command:
sh
Copy code
git push origin main
Example Workflow:

sh
Copy code
gh auth login
cd path/to/your/repository
gh pr list
gh pr checkout 42  # assuming PR #42
# Review the changes
gh pr merge 42 --merge  # merges with a merge commit
That's it! The pull request should now be merged.
